src/blazingmq/_ext.pyx

```diff
@@ -164,15 +165,44 @@ cdef class Session:
         on_message=None,
         broker not None: bytes = b'tcp://localhost:30114',
         message_compression_algorithm not None=_enums.CompressionAlgorithmType.NONE,
-        timeout: Optional[int|float] = None,
+        num_processing_threads: Optional[int] = None,
+        blob_buffer_size: Optional[int] = None,
+        channel_high_watermark: Optional[int] = None,
+        event_queue_watermarks: Optional[tuple[int,int]] = None,
+        stats_dump_interval: Optional[int|float] = None,
+        connect_timeout: Optional[int|float] = None,
```

Comment:
Passing in `class Timeouts` will make for a cleaner interface, unless there is some difficulty due to using `cdef` etc.